File: 2020/1/1.1.cpp
#include <fstream>
#include <iostream>
#include <unordered_set>

int main() {
  int x;
  std::ifstream input("1.txt");
  std::unordered_set<int> s;
  if (!input) {
    std::cerr << "Couldn't open 1.txt" << std::endl;
    return -1;
  }
  while (input >> x) {
    int y = 2020 - x;
    if (s.find(y) != s.end()) {
      std::cout << x << " * " << y << " = " << x*y << std::endl;
      return 0;
    }
    s.insert(x);
  }
  std::cerr << "Couldn't find pair summing up to 2020 in 1.txt" << std::endl;
  return -1;
}

File: 2016/1/1.1.py
input = open("input.txt",'r').read().strip().split(', ')

di = [ 0, 1, 0, -1 ]
dj = [ 1, 0, -1, 0 ]

i,j,d = 0,0,3
for step in input:
    d = (d+1 if step[0]=='R' else d+3) % 4
    s = int(step[1:])
    i,j = i+s*di[d], j+s*dj[d]
print(i,j, i+j)
